subsys/bluetooth/controller/ll_sw/ull.c

```diff
@@ -2464,21 +2464,56 @@ static uint8_t tx_cmplt_get(uint16_t *handle, uint8_t *first, uint8_t last)
 			struct node_tx_iso *tx_node_iso;
 			struct pdu_data *p;
 
+			/* NOTE: tx_cmplt_get() is permitted to be called
+			 *       multiple times before the tx_ack queue which is
+			 *       associated with Rx queue is changed by the
+			 *       dequeue of Rx node.
+			 *
+			 *       Tx node is released early without waiting for
+			 *       any dependency on Rx queue. Released Tx node
+			 *       reference is overloaded to store the Tx
+			 *       fragment count.
+			 *
+			 *       A hack is used here that depends on the fact
+			 *       that memory addresses have a value greater than
+			 *       0xFF, to determined if a node Tx has been
+			 *       released in a prior iteration of this function.
+			 */
 			tx_node_iso = tx->node;
 			p = (void *)tx_node_iso->pdu;
+			/* TODO: We may need something more advanced for framed */
+			if ((((uint32_t)tx_node_iso & ~0xFF) &&
+			     ((p->ll_id == PDU_CIS_LLID_COMPLETE_END) ||
+			      (p->ll_id == PDU_BIS_LLID_COMPLETE_END))) ||
+			    ((uint32_t)tx_node_iso & 0xFF)) {
+				uint8_t fragments;
+
+				/* We must count each SDU HCI fragment */
+				if ((uint32_t)tx_node_iso & ~0xFF) {
+					if (IS_ADV_ISO_HANDLE(tx->handle)) {
+						fragments = 1U;
+					} else {
+						fragments = tx_node_iso->sdu_fragments;
+					}
+				} else {
+					fragments = (uint32_t)tx_node_iso;
+				}
 
-			if (IS_ADV_ISO_HANDLE(tx->handle)) {
-				/* FIXME: ADV_ISO shall be updated to use ISOAL for
-				 * TX. Until then, assume 1 node equals 1 fragment.
-				 */
-				cmplt += 1;
-			} else {
-				/* We count each SDU fragment completed by this PDU */
-				cmplt += tx_node_iso->sdu_fragments;
```

Comment:
@mtpr-ot This is not a revert of previous changes in ull.c. Do you mean the comments missing?